de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/ProjectNegotiationCollector.java

```diff
@@ -0,0 +1,123 @@
+package de.fu_berlin.inf.dpp.negotiation;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Random;
+
+import org.apache.log4j.Logger;
+
+import de.fu_berlin.inf.dpp.filesystem.IProject;
+import de.fu_berlin.inf.dpp.filesystem.IResource;
+import de.fu_berlin.inf.dpp.session.ISarosSession;
+
+/**
+ * This class collects resources that should be added to a session to add them
+ * at once in preparation of a new single project negotiation. The reason is to
+ * prevent multiple concurrent running project negotiations per user.
+ * 
+ * This class allows to hold the reference to a thread that should handle this
+ * new project negotiation.
+ */
+public class ProjectNegotiationCollector {
+    private static final Logger log = Logger
+        .getLogger(ProjectNegotiationCollector.class);
+
+    private final Map<IProject, List<IResource>> mapping = new HashMap<IProject, List<IResource>>();
+
+    private Thread worker = null;
+
+    /**
+     * Add project resource mappings for the next project negotiation.
+     * 
+     * @param projectResourcesMapping
+     *            project resource mappings to add
+     */
+    public synchronized void add(
+        Map<IProject, List<IResource>> projectResourcesMapping) {
+        for (Entry<IProject, List<IResource>> mapEntry : projectResourcesMapping
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Reduce the total number of break and continue statements in this loop to use at most one. ([squid:S135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=272&issues=AWb1y62KW3Y69bz5Xor5&open=AWb1y62KW3Y69bz5Xor5)

[ISSUE_KEY]: AWb1y62KW3Y69bz5Xor5